packages/pilot/src/containers/Reprocess/index.js

```diff
@@ -34,143 +34,177 @@ const validateOnRestart = ({ onRestart, stepStatus }, propName) => {
-      transaction: {
-        amount,
-        card: {
-          first_digits, // eslint-disable-line camelcase
-          holder_name, // eslint-disable-line camelcase
-          last_digits, // eslint-disable-line camelcase
-        },
-        payment: {
-          installments,
-        },
-      },
-    } = this.props
+const buildStep = (stepName, translation) => applySpec({
+  id: always(stepName),
+  status: prop(stepName),
+  title: () => translation(`pages.reprocess.${stepName}`),
+})
 
-    return (
-      <Modal
-        isOpen={isOpen}
-        loading={loading}
-        onRequestClose={onCancel}
-      >
-        <ModalTitle
-          closeIcon={<IconClose height={16} width={16} />}
-          onClose={onCancel}
-          title={t('pages.reprocess.title')}
-        />
-        <ModalContent>
-          <ModalSection>
-            <Steps
-              status={[
-                {
-                  id: 'confirmation',
-                  status: formatStepStatus(stepStatus.confirmation),
-                },
-                {
-                  id: 'result',
-                  status: formatStepStatus(stepStatus.result),
-                },
-              ]}
-              steps={[
-                {
-                  id: 'confirmation',
-                  title: t('pages.reprocess.confirmation'),
-                },
-                {
-                  id: 'result',
-                  title: t('pages.reprocess.result'),
-                },
-              ]}
-            />
-          </ModalSection>
-        </ModalContent>
-        {stepStatus
-          && stepStatus.result
-          && stepStatus.result !== 'pending'
-          ? (
-            <Result
-              onCopyIdClick={onCopyId}
-              onRestart={onRestart}
-              onViewTransactionClick={onViewTransaction}
-              status={stepStatus.result}
-              statusMessage={statusMessage}
-              t={t}
-            />
-          )
-          : (
-            <Form
-              amount={amount}
-              cardFirstDigits={first_digits} // eslint-disable-line camelcase
-              cardLastDigits={last_digits} // eslint-disable-line camelcase
-              error={
-                stepStatus.confirmation === 'error'
-                  ? statusMessage
-                  : ''
-              }
-              holderName={holder_name} // eslint-disable-line camelcase
-              loading={loading}
-              installments={installments}
-              onCancel={onCancel}
-              onConfirm={onConfirm}
-              t={t}
-            />
-          )
-        }
-      </Modal>
+const buildSteps = translation => juxt([
+  buildStep('identification', translation),
+  buildStep('confirmation', translation),
+  buildStep('result', translation),
+])
+
+const getSteps = allowReprocessWithoutAntifraud => uncurryN(2,
+  translation => pipe(
+    buildSteps(translation),
+    when(
+      () => !allowReprocessWithoutAntifraud,
```

Comment:
aqui dave pra usar aquele `always` maroto ex:
```suggestion
      always(!allowReprocessWithoutAntifraud)
```

Reply:
Boa, ajustado!